processing/src/test/java/io/druid/query/topn/TopNQueryRunnerTest.java

```diff
@@ -2325,4 +2330,247 @@ public void testTopNTimeExtraction()
+                        new MinAggregatorFactory("minIndex", "index")
+                    )
+                )
+            )
+        )
+        .postAggregators(Arrays.<PostAggregator>asList(QueryRunnerTestHelper.addRowsIndexConstant))
+        .build();
+
+    Map<String, Object> map = Maps.newHashMap();
+    map.put("null_column", null);
+    map.put("rows", 1209L);
+    map.put("index", 503332.5071372986D);
+    map.put("addRowsIndexConstant", 504542.5071372986D);
+    map.put("uniques", QueryRunnerTestHelper.UNIQUES_9);
+    map.put("maxIndex", 1870.06103515625D);
+    map.put("minIndex", 59.02102279663086D);
+    List<Result<TopNResultValue>> expectedResults = Arrays.asList(
+        new Result<>(
+            new DateTime("2011-01-12T00:00:00.000Z"),
+            new TopNResultValue(
+                Arrays.asList(
+                    map
+                )
+            )
+        )
+    );
+    HashMap<String, Object> context = new HashMap<String, Object>();
+    TestHelper.assertExpectedResults(expectedResults, runner.run(query, context));
+  }
+
+  @Test
+  public void notNullDimension()
+  {
+    TopNQuery query = new TopNQueryBuilder()
+        .dataSource(QueryRunnerTestHelper.dataSource)
+        .granularity(QueryGranularity.DAY)
+        .dimension(QueryRunnerTestHelper.marketDimension)
+        .metric(QueryRunnerTestHelper.uniqueMetric)
+        .threshold(1000)
+        .intervals(QueryRunnerTestHelper.firstToThird)
+        .aggregators(QueryRunnerTestHelper.commonAggregators)
+        .build();
+
+    HashMap<String, Object> context = new HashMap<String, Object>();
+    Iterator<? extends Result> result = Sequences.toList(runner.run(query, context), Lists.<Result>newArrayList())
+                                                 .iterator();
+    while (result.hasNext()) {
+      TopNResultValue value = (TopNResultValue) result.next().getValue();
+      Assert.assertFalse(value.getValue().isEmpty());
+    }
+  }
+
+  @Test
+  public void nullDimension()
+  {
+    TopNQuery query = new TopNQueryBuilder()
+        .dataSource(QueryRunnerTestHelper.dataSource)
+        .granularity(QueryGranularity.DAY)
+        .dimension("null_column")
+        .metric(QueryRunnerTestHelper.uniqueMetric)
+        .threshold(1000)
+        .intervals(QueryRunnerTestHelper.firstToThird)
+        .aggregators(QueryRunnerTestHelper.commonAggregators)
+        .build();
+
+    HashMap<String, Object> context = new HashMap<String, Object>();
+    Iterator<? extends Result> result = Sequences.toList(runner.run(query, context), Lists.<Result>newArrayList())
+                                                 .iterator();
+
+    while (result.hasNext()) {
+      Result queryResult = result.next();
+      TopNResultValue value = (TopNResultValue) queryResult.getValue();
+      Assert.assertFalse(queryResult.toString(), value.getValue().isEmpty());
+    }
+  }
+
+  @Test
+  public void nullDimensionWithFilter()
```

Comment:
I copied the tests as it is from provided the issue, have modified the names, remove redundant checks and added verification against actual values.